### PR TITLE
Fixes #22917 - fix smart proxy validation on foreman tests

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

File: app/models/katello/concerns/smart_proxy_extensions.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def refresh
         validates :download_policy, inclusion: {
           :in => DOWNLOAD_POLICIES,
           :message => _("must be one of the following: %s") % DOWNLOAD_POLICIES.join(', ')
-        }
+        }, :if => lambda { |sp| sp.download_policy }
```
Comment (akofink) [inline app/models/katello/concerns/smart_proxy_extensions.rb]: Is this just a nil check?

**Comment by akofink** [comment]
I tried testing this in the console, and it's a bit strange:

```rb
[13] pry(main)> sp.download_policy = nil
=> nil
[14] pry(main)> sp
=> #<SmartProxy:0x00000010ab4270
 id: 1,
 name: "centos7-devel.akofink-desktop",
 url: "https://centos7-devel.akofink-desktop:9090",
 created_at: Wed, 11 Apr 2018 13:11:32 UTC +00:00,
 updated_at: Wed, 11 Apr 2018 13:11:33 UTC +00:00,
 expired_logs: "0",
 puppet_path: "/etc/puppetlabs/code/environments",
 download_policy: nil>
[15] pry(main)> sp.valid?
=> true
[16] pry(main)> sp
=> #<SmartProxy:0x00000010ab4270
 id: 1,
 name: "centos7-devel.akofink-desktop",
 url: "https://centos7-devel.akofink-desktop:9090",
 created_at: Wed, 11 Apr 2018 13:11:32 UTC +00:00,
 updated_at: Wed, 11 Apr 2018 13:11:33 UTC +00:00,
 expired_logs: "0",
 puppet_path: "/etc/puppetlabs/code/environments",
 download_policy: "on_demand">
```

It's like the validation is setting the download policy?

**Comment by jlsherrill** [comment]
it is, we use a before_validation to set the default download_policy

File: app/models/katello/concerns/smart_proxy_extensions.rb (hunk 1 above)
Comment (jlsherrill) [reply app/models/katello/concerns/smart_proxy_extensions.rb]: yep!  so :if => :download_policy should be sufficient!

**Comment by akofink** [comment]
If we set it prior to validation, why would this value ever be nil?

**Pull request closed without merging**